.github/workflows/docker-build.yml

```diff
@@ -21,40 +22,41 @@ jobs:
     permissions:
       contents: write
       packages: write
-
     steps:
       - uses: "actions/checkout@v3"
-
-      - name: set up go
-        uses: actions/setup-go@v3
-        with:
-          go-version: ${{ env.GO_VERSION }}
-
       - name: Docker meta
         id: meta
         uses: docker/metadata-action@v4
         with:
           images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
           tags: |
             type=sha
-            type=semver,pattern={{major}}.{{minor}}.{{patch}}
-            type=semver,pattern={{major}}.{{minor}}
-            type=semver,pattern={{major}}
-
+            type=semver,pattern={{version}}
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v2
-
       - name: Login to GHCR
         uses: docker/login-action@v2
         with:
           registry: ghcr.io
           username: ${{ github.repository_owner }}
           password: ${{ secrets.GITHUB_TOKEN }}
-
-      - name: Build and push
+      - name: Build and Push amd64 Image
+        if: |
+          github.ref != 'refs/heads/main' &&
+          !startsWith(github.ref, 'refs/tags/v')
+        uses: docker/build-push-action@v3
+        with:
+          platforms: linux/amd64
+          push: true
```

Comment:
Do we need to push on all Pull Request commits? That seems like like a lot. Can't we just verify that the image builds? 

Or is the reason for pushing on all PR commits to run the smoke test? If so, maybe we can build the run the tests without pushing the image and cluttering up the image repo